core/src/main/java/com/yahoo/oak/ReferenceCodec.java

```diff
@@ -77,13 +77,47 @@ protected ReferenceCodec(int firstBitSizeLimit, int secondBitSizeLimit, int thir
         this.thirdMask  = mask(this.thirdBitSize);
     }
 
-    /*------- Static helpers -------*/
+    /**
+     * Construct the codec setting the number of bits to be used for each parameter.
+     * This instance would combine only 2 integers inside long.
+     * The bit sizes of 1 parameters can be given and the third is always the remaining bits.
+     * @param firstBitSizeLimit an upper limit on the size of the first parameter (exclusive)
+     *                          if invalid calculate according to other two limits
+     * @param secondBitSizeLimit an upper limit on the size of the second parameter (exclusive)
+     *                          if invalid calculate according to other two limits
+     */
+    protected ReferenceCodec(int firstBitSizeLimit, int secondBitSizeLimit) {
```

Comment:
This can be implemented by calling the other constructor as follows:
```
this(firstBitSizeLimit, firstBitSizeLimit, 0);
```

Reply:
Not straight forward, then the other constructor needs to be rewritten. I think the code is clearer as it is now. Not fixed, please reopen if you find it wrong.